lib/resty/jwt.lua

```diff
@@ -824,12 +826,17 @@ function _M.verify_jwt_obj(self, secret, jwt_obj, ...)
   end
 
   -- if jwe, invoked verify jwe
-  if jwt_obj[str_const.header][str_const.enc]  then
+  if jwt_obj.typ == str_const.JWE then
```

Comment:
As you remove the check of `enc`, you should add the check in verify_jew_obj

Reply:
I will fix this problem together in the next PR